datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/analytics/resolver/GetHighlightsResolver.java

```diff
@@ -37,44 +39,70 @@ public final List<Highlight> get(DataFetchingEnvironment environment) throws Exc
-        _analyticsService.getHighlights(_analyticsService.getUsageIndexName(), Optional.of(dateRange),
-            ImmutableMap.of(), ImmutableMap.of(), Optional.of("browserId"));
-
-    int weeklyActiveUsersLastWeek =
-        _analyticsService.getHighlights(_analyticsService.getUsageIndexName(), Optional.of(dateRangeLastWeek),
-            ImmutableMap.of(), ImmutableMap.of(), Optional.of("browserId"));
+  private Highlight getTimeBasedHighlight(
+          final String title,
+          final String changeString,
+          final DateTime endDateTime,
+          final Function<DateTime, DateTime> periodStartFunc
+  ) {
+    DateTime startDate = periodStartFunc.apply(endDateTime);
+    DateTime timeBeforeThat = periodStartFunc.apply(startDate);
+    DateRange dateRangeThis = new DateRange(
+            String.valueOf(startDate.getMillis()),
+            String.valueOf(endDateTime.getMillis())
+    );
+    DateRange dateRangeLast = new DateRange(
+            String.valueOf(timeBeforeThat.getMillis()),
+            String.valueOf(startDate.getMillis())
+    );
+
+    int activeUsersThisRange =  _analyticsService.getHighlights(
+            _analyticsService.getUsageIndexName(),
+            Optional.of(dateRangeThis),
+            ImmutableMap.of(),
+            ImmutableMap.of(),
+            Optional.of("browserId")
+    );
+    int activeUsersLastRange =  _analyticsService.getHighlights(
+            _analyticsService.getUsageIndexName(),
+            Optional.of(dateRangeLast),
+            ImmutableMap.of(),
+            ImmutableMap.of(),
+            Optional.of("browserId")
+    );
 
     String bodyText = "";
-    if (weeklyActiveUsersLastWeek > 0) {
-      Double percentChange =
-          (Double.valueOf(weeklyActiveUsers) - Double.valueOf(weeklyActiveUsersLastWeek)) / Double.valueOf(
-              weeklyActiveUsersLastWeek) * 100;
+    if (activeUsersLastRange > 0) {
+      double percentChange = (double) (activeUsersThisRange - activeUsersLastRange) 
+        / (double) activeUsersLastRange * 100;
 
       String directionChange = percentChange > 0 ? "increase" : "decrease";
 
       bodyText = Double.isInfinite(percentChange) ? ""
-          : String.format("%.2f%% %s from last week", percentChange, directionChange);
+              : String.format(changeString, percentChange, directionChange);
     }
+    return Highlight.builder().setTitle(title).setValue(activeUsersThisRange).setBody(bodyText).build();
+  }
 
-    highlights.add(Highlight.builder().setTitle(title).setValue(weeklyActiveUsers).setBody(bodyText).build());
+  /**
+   * TODO: Config Driven Charts Instead of Hardcoded.
+   */
+  private List<Highlight> getHighlights() {
+    final List<Highlight> highlights = new ArrayList<>();
+
+    DateTime endDate = DateTime.now();
+    highlights.add(getTimeBasedHighlight(
+            "Weekly Active Users",
+            "%.2f%% %s from last week",
+            endDate,
+            (date) -> date.minusWeeks(1)
```

Comment:
Minor: Why make it complex with a function instead of just passing startTime and endTime into the function?

Reply:
Or is this about passing the Interval? That's my guess